File: chap14/fig14-5-left.R
library(dplyr)
library(tidyr)
library(ggplot2)
library(latex2exp)

N <- 50
K <- 6
I <- 120
fit_MAP <- readRDS('output/result-model14-3.RDS')
phi_est <- fit_MAP$draws('phi', format='df') %>%
  select(starts_with('phi[')) %>% 
  pivot_longer(everything(), names_to='Parameter') %>% 
  extract(col='Parameter', into=c('Pattern', 'ItemID'), regex='phi\\[(\\d+),(\\d+)\\]', convert=TRUE) %>% 
  mutate(Pattern_label=paste0('k = ', Pattern))

p <- ggplot(data=phi_est, aes(x=ItemID, y=value)) +
  theme_bw(base_size=18) +
  theme(axis.text.x=element_text(angle=65, vjust=1, hjust=1)) +
  facet_wrap(~Pattern_label, ncol=3) +
  coord_flip() +
  scale_x_reverse(breaks=c(1, seq(20, 120, 20))) +
  geom_bar(stat='identity') +
  labs(y=TeX('$\\phi\\[k,i\\]$'))
ggsave(p, file='output/fig14-5-left.png', dpi=300, w=7, h=5)

File: chap14/fig14-5-left.py
import pandas
import cmdstanpy
from plotnine import ggplot, aes, theme_bw, theme, element_text, facet_wrap, coord_flip, scale_x_reverse, geom_bar, labs

d = pandas.read_csv('input/data-matrix-decomp.csv')
N = 50
K = 6
I = 120

fit_MAP = cmdstanpy.from_csv('output/result-model14-3')
phi_est = pandas.DataFrame(fit_MAP.stan_variable('phi'), columns=range(1, I+1))
phi_est = pandas.concat([pandas.DataFrame({'Pattern':range(1, K+1)}), phi_est], axis=1)
d_plot = pandas.melt(phi_est, id_vars='Pattern', var_name='ItemID')
d_plot['ItemID'] = d_plot.ItemID.astype('int32')

p = (ggplot(d_plot, aes('ItemID', 'value'))
    + theme_bw(base_size=18)
    + theme(axis_text_x=element_text(angle=65, vjust=1, hjust=1))
    + facet_wrap('~ Pattern', ncol=3)
    + coord_flip()
    + scale_x_reverse(breaks=[1, *range(20, 140, 20)])
    + geom_bar(stat='identity')
    + labs(y='$\phi[k,i]$')
)
p.save(filename='output/fig14-5-left.py.png', dpi=300, width=7, height=5)
